InnerEye/ML/deep_learning_config.py

```diff
@@ -470,6 +480,19 @@ def use_gpu(self) -> bool:
         from InnerEye.ML.utils.ml_util import is_gpu_available
         return is_gpu_available()
 
+    @property
+    def num_gpus_per_node(self) -> int:
+        """
+        Computes the number of gpus to use for each node: either the number of gpus available on the device
+        or restrict it to max_num_gpu, whichever is smaller. Returns 0 if running on a CPU device.
+        """
+        import torch
+        num_gpus = torch.cuda.device_count() if self.use_gpu else 0
+        logging.info(f"Number of available GPUs: {num_gpus}")
+        if 0 <= self.max_num_gpus < num_gpus:
```

Comment:
If there are not as many gpus as the user wants, do we still want to let the user know about that?

Reply:
added a warning